### PR TITLE
Add settings API

### LINKED ISSUE #1932: Dashboard GraphQL endpoint for site settings
We need to be able to fetch (and change where currently possible) things like:
- [x] the name and description of the current site
- [x] the list of supported countries
- [x] the list of currencies
- [x] the list of available tax rates
- [x] authorization keys
- [x] mutation to assign `top_menu` and `bottom_menu`
- [x] default currency

### LINKED ISSUE #2705: Add site settings mutations
### What I'm trying to achieve
To be able to update site settings, like shop name or domain.

### Describe a proposed solution
Add missing GraphQL endpoints.

### PR DESCRIPTION
**Pull request by artursmet**
I want to merge this change because it resolves #2705 

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).

### CODE REVIEW

File: saleor/graphql/site/mutations.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+from ..core.mutations import BaseMutation
+from ..core.types.common import WeightUnitsEnum
+from .types import SiteSettings
+
+
+class SiteSettingsInput(graphene.InputObjectType):
+    domain = graphene.String(description='Domain name for shop')
+    header_text = graphene.String(description='Header text')
+    description = graphene.String(description='SEO description')
+    top_menu = graphene.ID(
```
Comment (maarcingebala) [inline saleor/graphql/site/mutations.py]: Assigning storefront menus is covered in PR #2721 - I've added dedicated mutations for that.

File: saleor/graphql/site/mutations.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        description='Menu to use on site top')
+    bottom_menu = graphene.ID(
+        description='Menu to use on site bottom')
+    include_taxes_in_prices = graphene.Boolean(
+        description='Include taxes in prices')
+    display_gross_prices = graphene.Boolean(
+        description='Display prices with tax in store')
+    track_inventory_by_default = graphene.Boolean(
+        description='Enable inventory tracking')
+    homepage_collection = graphene.ID(
```
Comment (maarcingebala) [inline saleor/graphql/site/mutations.py]: Maybe this should be done in a dedicated mutation?

Comment (maarcingebala) [reply saleor/graphql/site/mutations.py]: Maybe as a rule of thumb, we should only put here simple settings such as booleans or strings, while having dedicated mutations for operations that manipulate objects or relationships?

File: saleor/graphql/shop/mutations.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+        if errors:
+            return ShopSettingsUpdate(errors=errors)
+        instance.save()
+        return ShopSettingsUpdate(shop=Shop(), errors=errors)
+
+
+class ShopDomainUpdate(BaseMutation):
+    class Arguments:
+        domain = graphene.String(description='Domain name for shop')
```
[the remark this answers is not in the document]
Comment (artursmet) [reply saleor/graphql/shop/mutations.py]: site name is not used anywhere I think

File: saleor/graphql/shop/mutations.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    include_taxes_in_prices = graphene.Boolean(
+        description='Include taxes in prices')
+    display_gross_prices = graphene.Boolean(
+        description='Display prices with tax in store')
+    track_inventory_by_default = graphene.Boolean(
+        description='Enable inventory tracking')
+    default_weight_unit = WeightUnitsEnum(description='Default weight unit')
+
+
+class ShopNavigationInput(graphene.InputObjectType):
```
[the remark this answers is not in the document]
Comment (artursmet) [reply saleor/graphql/shop/mutations.py]: Good catch!

File: saleor/graphql/shop/mutations.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (maarcingebala) [reply saleor/graphql/shop/mutations.py]: It is used as the storefront's HTML `title`.

[the remark this answers is not in the document]
Comment (artursmet) [reply saleor/graphql/shop/mutations.py]: Right. I've added support for changing site name.

**Pull request merged**